packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.component.tsx

```diff
@@ -1,188 +1,154 @@
-  const obs = findObs(encounter, obsConcept);
-
-  if (isTrueFalseConcept) {
-    if (obs.value.uuid == 'cf82933b-3f3f-45e7-a5ab-5d31aaee3da3') {
-      return 'Yes';
-    } else {
-      return 'No';
-    }
-  }
-  if (!obs) {
-    return '--';
-  }
-  if (isDate) {
-    return moment(obs.value).format('DD-MMM-YYYY');
-  }
-  if (typeof obs.value === 'object') {
-    return obs.value.names?.find((conceptName) => conceptName.conceptNameType === 'SHORT')?.name || obs.value.name.name;
-  }
-  return obs.value;
-}
-
-export const EncounterTile: React.FC<EncounterTileProps> = ({
-  patientUuid,
-  encounterUuid,
-  columns,
-  headerTitle,
-  description,
-  filter,
-  tileStyle,
-}) => {
-  const { t } = useTranslation();
-  const [allRows, setAllRows] = useState([]);
+export const EncounterTile: React.FC<EncounterTileProps> = ({ patientUuid, columns, headerTitle }) => {
+  const [lastEncounter, setLastEncounter] = useState(null);
   const [isLoading, setIsLoading] = useState(true);
-  const [counter, setCounter] = useState(0);
 
-  const loadRows = useCallback(
-    (encounterType) => {
-      const query = `encounterType=${encounterType}&patient=${patientUuid}`;
-      openmrsFetch(`/ws/rest/v1/encounter?${query}&v=${encounterRepresentation}`).then(({ data }) => {
-        if (data.results?.length > 0) {
-          let sortedEncounters = data.results.sort(
-            (firstEncounter, secondEncounter) =>
-              new Date(secondEncounter.encounterDatetime).getTime() -
-              new Date(firstEncounter.encounterDatetime).getTime(),
+  useEffect(() => {
+    columns.map((column) => {
+      const lastEncounter = fetchPatientLastEncounter(column.encounterUuid);
+      lastEncounter.then((value) => {
+        if (column.isARTDateConcept) {
+          column.obsValue = getObsFromEncounter(
+            value,
+            getARTDateConcept(
+              value,
+              artTherapyDateTime_UUID,
+              switchDateUUID,
+              substitutionDateUUID,
+              artStopDateUUID,
+              dateRestartedUUID,
+            ),
+            column.isConceptDate,
           );
+        } else {
+          column.obsValue = getObsFromEncounter(value, column.concept, column.isConceptDate);
+        }
 
-          if (filter) {
-            sortedEncounters = sortedEncounters.filter((encounter) => filter(encounter));
+        if (column.summaryConcept) {
+          if (column.isSummaryDaysCalculation) {
+            let summaryDate = getObsFromEncounter(value, column.summaryConcept);
+            if (summaryDate !== '--') {
+              const dateDifference = new Date().getTime() - new Date(summaryDate).getTime();
+              const totalDays = Math.floor(dateDifference / (1000 * 3600 * 24));
+              column.summaryObsValue = `${totalDays} days`;
+            }
+          } else {
+            column.summaryObsValue = getObsFromEncounter(value, column.summaryConcept, column.isConceptSummaryDate);
```

Comment:
This component needs to be reusable, that means it should be generic and not have specific logic. These need to be moved away from this component and added to the specific logic added to the specific implementations